rascal-vscode-extension/src/extension.ts

```diff
@@ -242,7 +236,6 @@ async function startTerminal(uri: vscode.Uri | undefined, ...extraArgs: string[]
         uri = undefined;
     }
     const rascal = await rascalClient;
-    await rascal.onReady();
```

Comment:
wasn't this essential? no source code comment to claim so, but why is it gone and what can happen if we remove this?

Reply:
It's discussed here: https://github.com/Microsoft/vscode-languageserver-node#3170-protocol-800-json-rpc-800-client-and-800-server

Roughly, the `onReady` stuff was bad design, you can no just register/send stuff before the client is connected, and the library will take care of the rest. This simplified some of our init code.

Reply:
It may also be the trigger for the infinite recursion you were talking about earlier this week?

Reply:
nope, that's about setting the language of a text editor. this code is only run for the very first time when we start-up the lsp server.